Source/Core/Common/Config/Layer.h

```diff
@@ -41,34 +94,41 @@ class Layer
   virtual ~Layer();
 
   // Convenience functions
-  bool Exists(System system, const std::string& section_name, const std::string& key);
-  bool DeleteKey(System system, const std::string& section_name, const std::string& key);
+  bool Exists(const ConfigLocation& location) const;
+  bool DeleteKey(const ConfigLocation& location);
+  void DeleteAllKeys();
+
   template <typename T>
-  bool GetIfExists(System system, const std::string& section_name, const std::string& key, T* value)
+  T Get(const ConfigInfo<T>& config_info)
   {
-    if (Exists(system, section_name, key))
-      return GetOrCreateSection(system, section_name)->Get(key, value);
-
-    return false;
+    return Get<T>(config_info.location).value_or(config_info.default_value);
   }
 
-  virtual Section* GetSection(System system, const std::string& section_name);
-  virtual Section* GetOrCreateSection(System system, const std::string& section_name);
-
   template <typename T>
-  T Get(const ConfigInfo<T>& config_info)
+  std::optional<T> Get(const ConfigLocation& location)
   {
-    return GetOrCreateSection(config_info.location.system, config_info.location.section)
-        ->template Get<T>(config_info.location.key, config_info.default_value);
+    const std::optional<std::string>& str_value = m_map[location];
+    if (!str_value)
+      return std::nullopt;
+    return detail::TryParse<T>(*str_value);
   }
 
   template <typename T>
   void Set(const ConfigInfo<T>& config_info, const T& value)
   {
-    GetOrCreateSection(config_info.location.system, config_info.location.section)
-        ->Set(config_info.location.key, value);
+    Set<T>(config_info.location, value);
   }
 
+  template <typename T>
+  void Set(const ConfigLocation& location, const T& value)
+  {
+    m_is_dirty = true;
```

Comment:
Sorry I missed this during my previous reviews, but this will cause the layer to be considered dirty whenever this is called even when the value hasn't changed -- which will result in useless saves. What do you think of keeping the previous behaviour and only setting the flag if the value has actually changed?

Reply:
I originally had half-the-mind to remove it all together since not saving doesn't really save much time; updated to reflect previous behaviour anyway.